diff --git a/src/soundio/sounddevicenetwork.cpp b/src/soundio/sounddevicenetwork.cpp
index aff65b9dc0..8da832d6a9 100644
--- a/src/soundio/sounddevicenetwork.cpp
+++ b/src/soundio/sounddevicenetwork.cpp
@@ -461,7 +461,7 @@ void SoundDeviceNetwork::callbackProcessClkRef() {
 #if defined(__aarch64__)
         // Flush-to-zero on aarch64 is controlled by the Floating-point Control Register
         // Load the register into our variable.
-        int savedFPCR;
+        int64_t savedFPCR;
         asm volatile("mrs %[savedFPCR], FPCR"
                      : [ savedFPCR ] "=r"(savedFPCR));

diff --git a/src/soundio/sounddeviceportaudio.cpp b/src/soundio/sounddeviceportaudio.cpp
index 681d0942cc..97a4429420 100644
--- a/src/soundio/sounddeviceportaudio.cpp
+++ b/src/soundio/sounddeviceportaudio.cpp
@@ -935,7 +935,7 @@ int SoundDevicePortAudio::callbackProcessClkRef(
 #if defined(__aarch64__)
         // Flush-to-zero on aarch64 is controlled by the Floating-point Control Register
         // Load the register into our variable.
-        int savedFPCR;
+        int64_t savedFPCR;
         asm volatile("mrs %[savedFPCR], FPCR"
                      : [ savedFPCR ] "=r"(savedFPCR));

--
2.40.0

