diff --git a/src/library/basetrackcache.cpp b/src/library/basetrackcache.cpp
index 829b190865..d87ba2db70 100644
--- a/src/library/basetrackcache.cpp
+++ b/src/library/basetrackcache.cpp
@@ -210,7 +210,7 @@ bool BaseTrackCache::updateTrackInIndex(
             getTrackValueForColumn(pTrack, i, record[i]);
         }
         if (m_bIsCaching) {
-            replaceRecentTrack(std::move(trackId), std::move(pTrack));
+            replaceRecentTrack(std::move(trackId), pTrack);
         }
     } else {
         if (m_bIsCaching) {
--
2.40.0
