--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -498,9 +498,9 @@ elseif(GNU_GCC OR LLVM_CLANG)
   endif()
 endif()

-set(CMAKE_C_VISIBILITY_PRESET hidden)
-set(CMAKE_CXX_VISIBILITY_PRESET hidden)
-set(CMAKE_VISIBILITY_INLINES_HIDDEN TRUE)
+#set(CMAKE_C_VISIBILITY_PRESET hidden)
+#set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+#set(CMAKE_VISIBILITY_INLINES_HIDDEN TRUE)

 if(WIN32)
   # Add support for lib prefix on Windows
--
2.41.0

