--- mixxx-1.9.2/build/depends.py	2011-11-01 20:01:21.000000000 -0400
+++ mixxx-1.9.2/build/depends.py.norpath	2011-11-08 11:32:58.261269786 -0500
@@ -179,8 +179,6 @@
         compiling_on_104 = False
         if build.platform_is_osx:
             compiling_on_104 = (os.popen('sw_vers').readlines()[1].find('10.4') >= 0)
-        if not build.platform_is_windows and not (using_104_sdk or compiling_on_104):
-            build.env.Append(LINKFLAGS = "-Wl,-rpath,$QTDIR/lib")
 
         #QtSQLite DLL
         if build.platform_is_windows:
