diff -up mixxx-1.8.2/src/SConscript.env.norpath mixxx-1.8.2/src/SConscript.env
--- mixxx-1.8.2/src/SConscript.env.norpath	2010-12-06 00:07:57.000000000 +0100
+++ mixxx-1.8.2/src/SConscript.env	2011-01-21 15:30:52.455311000 +0100
@@ -837,9 +837,6 @@ if 'win' in platformString:
 else:
 	env.Append(CCFLAGS = Split(""" -pipe -Wall -W -g """)) # omghax
 	env.Append(LINKFLAGS = Split(""" -pipe -Wall -W -g"""))
-	if platformString != 'osx':
-		env.Append(LINKFLAGS = "-Wl,-rpath,$QTDIR/lib")
-	env.Append(LINKFLAGS = "-Wl,-rpath,$QTDIR/lib")
 	env.Append(CPPDEFINES = "_REENTRANT")
 
 #Uic these guys (they're moc'd automatically after this) - Generates the code for the QT UI forms
