--- mixxx/build/mixxx.py	2012-09-05 05:35:20 +0000
+++ mixxx/build/mixxx.py	2013-05-26 23:43:44 +0000
@@ -130,7 +130,7 @@
 
         # Validate the specified qtdir exists
         if not os.path.exists(qtdir):
-            logging.error("QT path does not exist or QT4 is not installed.")
+            logging.error("QT path (%s) does not exist or QT4 is not installed." % qtdir)
             logging.error("Please specify your QT path by running 'scons qtdir=[path]'")
             Script.Exit(1)
         # And that it doesn't contain qt3

