diff -ur mixxx-2.0.0/build/mixxx.py mixxx-2.0.0-patched/build/mixxx.py
--- mixxx-2.0.0/build/mixxx.py	2015-12-29 17:10:41.000000000 +0100
+++ mixxx-2.0.0-patched/build/mixxx.py	2016-02-05 15:22:30.638429571 +0100
@@ -162,7 +162,7 @@
 
         # Validate the specified qtdir exists
         if not os.path.isdir(qtdir):
-            logging.error("QT path does not exist or QT4 is not installed.")
+            logging.error("QT path (%s) does not exist or QT4 is not installed." % qtdir)
             logging.error(
                 "Please specify your QT path by running 'scons qtdir=[path]'")
             Script.Exit(1)
@@ -371,7 +371,7 @@
         if 'CC' in os.environ:
             self.env['CC'] = os.environ['CC']
         if 'CFLAGS' in os.environ:
-            self.env['CFLAGS'] += SCons.Util.CLVar(os.environ['CFLAGS'])
+            self.env['CCFLAGS'] += SCons.Util.CLVar(os.environ['CFLAGS'])
         if 'CXX' in os.environ:
             self.env['CXX'] = os.environ['CXX']
         if 'CXXFLAGS' in os.environ:
--- ./build/depends.py.orig	2016-02-09 04:30:54.860754912 +0000
+++ ./build/depends.py	2016-02-09 04:29:53.286383033 +0000
@@ -359,7 +359,6 @@ class Qt(Dependence):
             qtdir = build.env['QTDIR']
             framework_path = Qt.find_framework_libdir(qtdir, qt5)
             if os.path.isdir(framework_path):
-                build.env.Append(LINKFLAGS="-Wl,-rpath," + framework_path)
                 build.env.Append(LINKFLAGS="-L" + framework_path)
 
 
