diff -up mixxx-1.10.0/src/engine/engineshoutcast.cpp.gcc47 mixxx-1.10.0/src/engine/engineshoutcast.cpp
--- mixxx-1.10.0/src/engine/engineshoutcast.cpp.gcc47	2011-12-24 03:53:20.000000000 +0100
+++ mixxx-1.10.0/src/engine/engineshoutcast.cpp	2012-05-03 23:03:38.068426488 +0200
@@ -37,6 +37,7 @@
 #include <QDebug>
 #include <QMutexLocker>
 #include <stdio.h> // currently used for writing to stdout
+#include <unistd.h> // sleep
 
 
 /*
