--- a/build/mixxx.py	2015-12-29 16:10:41.000000000 +0000
+++ b/build/mixxx.py	2016-08-21 07:48:27.404004423 +0100
@@ -48,12 +48,14 @@
         if machine.lower() not in ['x86_64', 'x86', 'i686', 'i586',
                                    'alpha', 'hppa', 'mips', 'mipsel', 's390',
                                    'sparc', 'ia64', 'armel', 'armhf', 'hurd-i386',
+                                   'armv5tel', 'armv5tejl', 'armv6l', 'armv6hl',
+                                   'armv7l', 'armv7hl', 'armv7hnl',
                                    'sh3', 'sh4',
                                    'kfreebsd-amd64', 'kfreebsd-i386',
                                    'i486', 'i386', 'ppc', 'ppc64', 'powerpc',
                                    'powerpc64', 'powerpcspe', 's390x',
                                    'amd64', 'em64t', 'intel64', 'arm64',
-                                   'ppc64el']:
+                                   'ppc64le', 'aarch64']:
             raise Exception("invalid machine type")
 
         if toolchain not in ['gnu', 'msvs']:

